src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private static final int RANDOM_RANGE = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        if (random.nextInt(NUMBER_OF_FIGURES) == 0) {
+            return new Circle(random.nextInt(RANDOM_RANGE),
+                    colorSupplier.getRandomColor().toString());
+        }
+        if (random.nextInt(NUMBER_OF_FIGURES) == 1) {
+            return new IsoscelesTrapezoid(random.nextInt(RANDOM_RANGE),
+                   random.nextInt(RANDOM_RANGE), random.nextInt(RANDOM_RANGE),
+                    colorSupplier.getRandomColor().toString());
+        }
+        if (random.nextInt(NUMBER_OF_FIGURES) == 2) {
+            return new Rectangle(random.nextInt(RANDOM_RANGE), random.nextInt(RANDOM_RANGE),
+                    colorSupplier.getRandomColor().toString());
+        }
+        if (random.nextInt(NUMBER_OF_FIGURES) == 3) {
+            return new RightTriangle(random.nextInt(RANDOM_RANGE),random.nextInt(RANDOM_RANGE),
+                    colorSupplier.getRandomColor().toString());
+        }
+        if (random.nextInt(NUMBER_OF_FIGURES) == 4) {
+            return new Square(random.nextInt(RANDOM_RANGE),
+                    colorSupplier.getRandomColor().toString());
+        }
```

Comment:
Lets generate random index only ones, and than use switch-case construction:
``` Java
int randomIndex = random.nextInt(NUMBER_OF_FIGURES);
switch (randomIndex) {
  case 1: return getRandomCircle();
  case 2: return getRandomSquare();
 	.....
  default: ....
}
```

and we can create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability (Remember private methods should follow after public as helper ones)